test/dest-symlinks.js

```diff
@@ -270,6 +352,85 @@ describe('.dest() with symlinks', function() {
     ], done);
   });
 
+  // This could also be from a different Vinyl adapter
+  it('(*nix) receives a virtual symbolic directory and creates a symlink', function(done) {
+    if (isWindows) {
+      this.skip();
+      return;
+    }
+
+    var file = new File({
+      base: neInputBase,
+      path: neInputDirpath,
+      contents: null,
+      stat: {
+        isSymbolicLink: isSymbolicLink,
+      },
+    });
+
+    // `src()` adds this side-effect with `resolveSymlinks` option set to false
+    file.symlink = neInputDirpath;
+
+    function assert(files) {
+      var lstats = fs.lstatSync(neOutputDirpath);
+      var outputLink = fs.readlinkSync(neOutputDirpath);
+      var linkTargetExists = fs.existsSync(outputLink);
+
+      expect(files.length).toEqual(1);
+      // TODO: Am I testing all the correct stuff here?
+      expect(outputLink).toEqual(neInputDirpath);
+      expect(linkTargetExists).toEqual(false);
+      expect(lstats.isDirectory()).toEqual(false);
+      expect(lstats.isSymbolicLink()).toEqual(true);
+    }
+
+    pipe([
+      from.obj([file]),
+      vfs.dest(neOutputBase),
+      concat(assert),
+    ], done);
+  });
+
+  // This could also be from a different Vinyl adapter
+  it('(windows) receives a virtual sybolic directory and creates a junction', function(done) {
```

Comment:
@erikkemperman same as above but for Windows.

Reply:
Ha, I guess this test represents my problem, as well as your's @erikkemperman - it didn't create a junction because it couldn't determine that the virtual directory was supposed to be a directory.  However, your solution doesn't solve this case because we can't look it up on the filesystem.  Ugh, I don't know how to resolve this.